app/models/order.rb

```diff
@@ -1,36 +1,88 @@
 class Order < ActiveRecord::Base
-  # DB relationships
-  has_many :order_items
+  has_many :order_items, dependent: :destroy # should destroy all of the associated OrderItems if an Order is destroyed.
+  has_many :products, through: :order_items
+
+  attr_accessor :confirmed_payment
+
+  after_initialize do |order|
+    order.confirmed_payment = false
+  end
 
-  # validations helper regex
   # email regex from: http://rails-3-2.railstutorial.org/book/modeling_users#code-validates_format_of_email
-  # VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
-  VALID_STATUS_REGEX = /(pending)|(paid)|(complete)|(cancelled)/
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  VALID_BUYER_CARD_SHORT_REGEX = /\A\d{4}\z/
+
+  validates :status, presence: true, inclusion: { in: %w(pending paid complete canceled),
```

Comment:
I think it would make sense to create a constant or constants that correspond to these statuses because you use them later on